lib/src/facts/linux/networking_resolver.cc

```diff
@@ -71,4 +89,85 @@ namespace facter { namespace facts { namespace linux {
         return interface;
     }
 
+    void networking_resolver::read_routing_table()
+    {
+        auto ip_command = lth_exe::which("ip");
+        if (ip_command.empty()) {
+            LOG_DEBUG("Could not find the 'ip' command. Network bindings will not be populated from routing table");
+            return;
+        }
+
+        auto parse_route_line = [](string& line, std::vector<route>& routes) {
+            vector<boost::iterator_range<string::iterator>> parts;
+            boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+            if (parts.size() % 2 != 1) {
+                LOG_WARNING("Could not process routing table entry: Expected a destination followed by key/value pairs, got '{}'", line);
+                return true;
+            }
+
+            route r;
+            r.destination.assign(parts[0].begin(), parts[0].end());
+            // Iterate over key/value pairs and add the ones we care
+            // about to our routes entries
+            for (size_t i = 1; i < parts.size(); i += 2) {
+                std::string key(parts[i].begin(), parts[i].end());
+                if (key == "dev") {
+                    r.interface.assign(parts[i+1].begin(), parts[i+1].end());
+                }
+                if (key == "src") {
+                    r.source.assign(parts[i+1].begin(), parts[i+1].end());
+                }
+            }
+            routes.push_back(r);
+            return true;
+        };
+
+        lth_exe::each_line(ip_command, { "route", "show" }, [this, &parse_route_line](string& line) {
+            return parse_route_line(line, this->routes4);
+        });
+        lth_exe::each_line(ip_command, { "-6", "route", "show" }, [this, &parse_route_line](string& line) {
+            return parse_route_line(line, this->routes6);
+        });
+    }
+
+    void networking_resolver::populate_from_routing_table(networking_resolver::data& result) const
+    {
+        for (const auto& r : routes4) {
+            if (r.destination == "default" && result.primary_interface.empty()) {
+                 result.primary_interface = r.interface;
+            }
+            if (!r.source.empty()) {
+                auto iface = find_if(result.interfaces.begin(), result.interfaces.end(), [&](const interface& iface) {
+                    return iface.name == r.interface;
+                });
+                if (iface != result.interfaces.end()) {
+                    auto existing_binding = find_if(iface->ipv4_bindings.begin(), iface->ipv4_bindings.end(), [&](const binding& b) {
```

Comment:
This is `O(n^2)`, but currently we don't see the number of bindings being large enough to require additional code complexity and the overhead of setting up a `set`. /cc @Sharpie on whether that's a bad assumption.